service/src/main/java/com/epam/esm/validator/GiftCertificateValidator.java

```diff
@@ -0,0 +1,100 @@
+package com.epam.esm.validator;
+
+import com.epam.esm.dto.GiftCertificate;
+import com.epam.esm.dto.Tag;
+
+import java.math.BigDecimal;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * The type Gift certificate validator.
+ */
+public class GiftCertificateValidator {
```

Comment:
Util class with static methods can be replaced with some other instance, especially when you have Spring.